datalad/support/annexrepo.py

```diff
@@ -1972,7 +1972,9 @@ def merge_annex(self, remote=None):  # do not use anymore, use localsync()
 
     def sync(self, remotes=None, push=True, pull=True, commit=True,
              content=False, all=False, fast=False):
-        """Synchronize local repository with remotes
+        """This method is deprecated, use call_annex(['sync', ...]) instead.
+
+        Synchronize local repository with remotes
```

Comment:
I think we should develop a habit of this:
`warnings.warn("AnnexRepo.sync() is deprecated, use call_annex(['sync', ...]) instead.", DeprecationWarning)`

Reply:
I added such a warning.